### DISCUSSION TITLE
Semver ranges + docs issues

### DISCUSSION THREAD

**Discussion by bcomnes** [community/community#25804] [Actions]
<p>In the versioning docs for actions, its encouraged that major version ‘tags’ are moved around:</p>
<p><a href="https://help.github.com/en/github/automating-your-workflow-with-github-actions/about-actions#versioning-your-action" rel="nofollow noopener">https://help.github.com/en/github/automating-your-workflow-with-github-actions/about-actions#versioning-your-action</a></p>
<p>&gt; Move the major version tag (v1, v2, etc.) to point to the Git ref of the current release. For more information, see “<a href="https://git-scm.com/book/en/v2/Git-Basics-Tagging" rel="nofollow noopener">Git basics - tagging</a>.”</p>
<p>Generally tags should be immutible and branches are the primitive that shouldbe used on git labels that move around.  </p>
<p>It does appear major version branches work justs as well as tags, and are generally easier to manage and move around.  However the docs and examples still point people in the direction of tags.</p>
<p>Is there a reason for that? It seems like branches would serve better than tags in this case. </p>
<p>Furthurmore, having to manage version range tags is error prone.  Surea bot can do it, however common.js has already solved this problem with semver range markers.  Why not just support those? <code>~v1.0.0</code> and scrap the major verstion tag management. </p>
<p>Loving actions otherwise!</p>

**Comment by ethomson** [comment, marked as answer]
<p>Thanks for the feedback.  Indeed this same strategy can be accomplished with branches instead of tags, and if that aligns more with the way you want to handle release management, I don’t see any reason why you can’t.</p>
<p>Long-term, I agree, we need to unburden you from managing semver releases yourself.  We’re looking at ways to teach GitHub Actions itself about semantic versioning, instead of making you do all the work.</p>
<p>Appreciate your thoughts on this, and we’ll work to improve this in the future.</p>
